tlsutil/config.go

```diff
@@ -375,20 +375,17 @@ func (c *Configurator) check(config Config, pool *x509.CertPool, cert *tls.Certi
 
 	// Ensure we have a CA and cert if VerifyIncoming is set
 	if config.anyVerifyIncoming() {
-		autoEncryptMsg := " AutoEncrypt only secures the connection between client and server and doesn't affect incoming connections on the client."
 		if pool == nil {
-			errMsg := "VerifyIncoming set, and no CA certificate provided!"
-			if config.AutoEncryptTLS {
-				errMsg += autoEncryptMsg
-			}
-			return fmt.Errorf(errMsg)
+			// both auto-config and auto-encrypt while being able to provide the agents certificate and some ca certs require
+			// verifying the connection to the server. Therefore its an error if we ever want to verify incoming certificates
+			// but have no CA certs
```

Comment:
I am not sure I quite understand this comment. I understand that if we do not have a CA pool then it is impossible for us to verify incoming connections, but I am not sure how the fact that auto-config/auto-encrypt "require verifying the connection to the server" plays into that?

Reply:
Oh ok, I think I might understand now. If I have a Consul server running with `verify_incoming: true`, then I must provide a CA certificate on startup in order to even be able to bootstrap auto-config/auto-encrypt.

For a consul client, I don't think we have the same problems with `verify_incoming: true`, but we do still need to set a CA for `verify_outgoing: true` in order to requests certs to be generated.

Reply:
Might be more readable if we separate out the clause:

> both auto-config and auto-encrypt, while being able to provide the agents certificate and some ca certs, require verifying the connection to the server.

Reply:
I can reword it essentially auto_encrypt and auto_config should only be used with verify_outgoing set which would require CA certificates to be present.